frontend/appflowy_tauri/src/appflowy_app/stores/effects/document/document_controller.ts

```diff
@@ -124,11 +128,25 @@ export class DocumentController {
     };
   };
 
+  private composeDelta = (node: Node) => {
+    const delta = node.data.delta;
+    if (!delta) {
+      return;
+    }
+    // we use yjs to compose delta, it can make sure the delta is correct
+    // for example, if we insert a text at the end of the line, the delta will be [{ insert: 'hello' }, { insert: " world" }]
+    // but if we use yjs to compose the delta, the delta will be [{ insert: 'hello world' }]
+    const ydoc = new Y.Doc();
```

Comment:
How about we use `delta.js` to compose it? https://github.com/quilljs/delta

Reply:
It is not really composed of delta and op, it just composes delta1 and delta2
for example
node1 delta: [{ insert: 'hello' }] 
node2 delta: [{ insert: 'world' }] 
when we compose node1 and node2, we will get [{ insert: 'hello' }, { insert: " world" }]
but we want the delta will be like [{ insert: 'hello world' }]
so we just put them in a ytext and get this ytext's delta